### DISCUSSION TITLE
为什么表单自定义验证使用async await后，callback不能显示，而promise .then是可以的

### DISCUSSION THREAD

**Discussion by WonderfulEdge** [ant-design/ant-design#35383] [Q&A]
[![OZqm0f.md.png](https://s1.ax1x.com/2022/05/05/OZqm0f.md.png)](https://imgtu.com/i/OZqm0f)
 **这样使用async，await是失败的。**

[![OZLJ8H.md.png](https://s1.ax1x.com/2022/05/05/OZLJ8H.md.png)](https://imgtu.com/i/OZLJ8H)
 **这样使用promise，.then是成功的。**

**Comment by lich-yoo** [comment]
因为 async 函数返回 Promise，此时 callback api 就禁掉了。
第二个箭头函数返回 void，所以 callback 还有用，
v4 版本推荐用 promise

**Reply by WonderfulEdge** [reply]
谢谢！！
